src/style-spec/expression/index.js

```diff
@@ -87,7 +85,7 @@ function createExpression(expression: mixed,
             }
             return val;
         } catch (e) {
-            if (!options.isConvertedFunction && !warningHistory[e.message]) {
+            if (!warningHistory[e.message]) {
```

Comment:
Just noting, we'll possibly need something like this again if/when we reinstate legacy function => expression conversion.